File: N-Tier/AspCoreProject/softweb.service/Services/UserService.cs

using System;
using System.Collections.Generic;
using System.Linq;
using softweb.persistence;
using Softweb.Models;

namespace softweb.service
{
    public class UserService : IUserService
    {
        private UserDao userDao;

        public UserService()
        {
            userDao = new UserDao();
        }

        public UserService(UserDao mockContext)
        {
            userDao = mockContext;
        }



        public List<UserDto> GetUsers()
        {
            List<UserDto> result = userDao.Users.Select(u => new UserDto(u.Id, u.firstName, u.lastName)).ToList();
            return result;
        }
    }
}

File: N-Tier/NodeJsProject/softweb-service/services/UserService.ts

import IUserservice from "../services/IUserservice";
import UserDto from "softweb-common/src/Models/UserDto";

import UserDao from "softweb-persistence/src/UserDao";
import User from "softweb-persistence/src/entity/User";

export default class UserService implements IUserservice {
    userDao : UserDao;
    constructor(context? : UserDao){
        this.userDao = (context)?context:new UserDao();
    }

    public GetUsers(): UserDto[] {
        let users: User[] = this.userDao.Users();
        let result = (users== null)? null : users.map(u=><UserDto>({Id:u.id,FirstName:u.firstName,LastName:u.lastName}));
        return result;
    }    
    
    public Multipl(v1: number, v2: number): number {
        return v1 * v2;
    }

}